core/templates/dev/head/pages/admin/roles_tab/RoleGraphDirective.js

```diff
@@ -0,0 +1,251 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for showing static role graph.
+ */
+
+oppia.directive('roleGraph', [
```

Comment:
I mentioned that, if we want to do a readOnlyGraph, we should do it as a reusable directive and get the history tab in the editor to use it. Can we do that? Otherwise you're just creating a one-use thing by duplicating code, and that's less good. Also I would name this readOnlyGraph (or something that's not role-specific) and try to ensure that its interface is generic so that it can be reused by other things which need this.